src/misc.c

```diff
@@ -206,7 +205,14 @@ char *fgetll(char *s, size_t size, FILE *stream)
 		while (!new_cp) {
 			increase >>= 2;
 			if (increase < 0x10000)
+#ifndef _JOHN_MISC_NO_LOG
 				pexit("realloc");
+#else
+			{
+				fprintf(stderr, "error allocating memory in misc.c fgetll()\n");
+				exit(0);
```

Comment:
This is a bug, you are not allowed to call exit. You should use error() or error_msg(), or why not the original pexit although with a more decriptive message.

Reply:
Because the ONLY reason we placed the _JOHN_MISC_NO_LOG there was due to the pexit code.  NOTE prior to this change, the ENTIRE function was removed.  WITH this change, the  function is there, I can test it with the TS, but the exit is changed.  Why the concern?  This code would never be compiled in for a normal john build. It would never have the _JOHN_MISC_NO_LOG code set.

Reply:
Oh, right. I missed the *n* if `#ifndef`!